crates/red_knot_test/src/lib.rs

```diff
@@ -141,10 +141,23 @@ fn run_test(db: &mut db::Db, test: &parser::MarkdownTest) -> Result<(), Failures
                 Ok(type_diagnostics) => type_diagnostics,
                 Err(info) => {
                     let mut by_line = matcher::FailuresByLine::default();
-                    by_line.push(
-                        OneIndexed::from_zero_indexed(0),
-                        info.info.split('\n').map(String::from).collect(),
-                    );
+                    let mut messages = vec![];
+                    match info.location {
+                        Some(location) => messages.push(format!("panicked at {location}")),
+                        None => messages.push("panicked at unknown location".to_string()),
+                    };
+                    match info.payload {
+                        Some(payload) => messages.push(payload),
+                        // Mimic the default panic hook's rendering of the panic payload if it's
+                        // not a string.
+                        None => messages.push("Box<dyn Any>".to_string()),
```

Comment:
So instead we just mimic the output of the default panic hook here if the payload is not a string.  (Note that neither us nor `std::panic` can print out anything more meaningful, since you can only downcast `Any` to specific known types, not to traits like `Display` or `Debug`)